pkg/kubectl/cmd/config/config.go

```diff
@@ -429,6 +430,10 @@ func getConfigFromFileOrDie(filename string) *clientcmdapi.Config {
 		return clientcmdapi.NewConfig()
 	}
 
+	if config.Clusters == nil {
+		config.Clusters = map[string]*clientcmdapi.Cluster{}
```

Comment:
I believe that the codec overrides whatever values are in the object.  This is what I would expect from an API standpoint -- if I point the codec at an empty file, which is valid yaml, i'd expect to get a zeroed object back.